### ISSUE TITLE
Add SourceLink support

### ISSUE THREAD

**Issue by gitfool** [tmenier/Flurl#513]
[SourceLink](https://github.com/dotnet/sourcelink) is standard now and would make it easy to debug code using this awesome package!

**Comment by tmenier** [comment]
Ready in prerelease 5: https://www.nuget.org/packages/Flurl.Http/3.0.0-pre5

**Comment by tmenier** [comment]
@gitfool Do we still need SourceLink/embedded PDBs now that symbols packages are being published? If it's effectively 2 ways to do the same thing then I'd really like to undo this. The most popular [request](https://github.com/tmenier/Flurl/issues/517) for 4.0 is removing Newtonsoft in favor of System.Text.Json, and a reason often cited is to reduce Flurl's footprint in things like Blazor and Xamarin apps. PDBs and the SourceLink dependency add to that footprint too, maybe not as much but people are quite sensitive about it and my goal is to keep the package as small and dependency-free as possible.

Basically I'm proposing we undo most of #514. Thoughts?

**Comment by gitfool** [comment]
You don't need to worry about that and shouldn't change anything. SourceLink is still configured and points back to the GitHub repository. This keeps the symbols fairly small plus the symbols are now in a separate NuGet package, so should only downloaded on demand. Note the differences in the NuGet packages before and after the `snupkg` change.

`Flurl.Http.nupkg` (3.2.3):
![image](https://user-images.githubusercontent.com/750121/170778698-58532b67-3489-49c7-b86e-45c98609ef5c.png)

`Flurl.Http.nupkg` (3.2.4):
![image](https://user-images.githubusercontent.com/750121/170778715-246b7e7d-6460-4404-a48e-b6c5d638ed57.png)

`Flurl.Http.snupkg` (3.2.4):
![image](https://user-images.githubusercontent.com/750121/170779420-77cb65c7-3952-4067-8b22-df34682e1482.png)

**Comment by tmenier** [comment]
Thanks for the quick response, I'll go forward without worrying about it.

Good point about a 6.0 target. Very relevant too - it appears the netstandard2.0 target will require the System.Text.Json NuGet package, which won't make those wanting a smaller footprint particularly happy! (I'll skip 5.0 since it's out of support already.)
